Firestore/core/test/unit/remote/remote_event_test.cc

```diff
@@ -564,6 +563,108 @@ TEST_F(RemoteEventTest, ExistenceFilterMismatchClearsTarget) {
   ASSERT_EQ(event.document_updates().size(), 0);
 }
 
+TEST_F(RemoteEventTest, ExistenceFilterMismatchWithBloomFilterSuccess) {
+  std::unordered_map<TargetId, TargetData> target_map = ActiveQueries({1, 2});
+
+  MutableDocument doc1 = Doc("docs/1", 1, Map("value", 1));
+  auto change1 = MakeDocChange({1}, {}, doc1.key(), doc1);
+  MutableDocument doc2 = Doc("docs/2", 2, Map("value", 2));
+  auto change2 = MakeDocChange({1}, {}, doc2.key(), doc2);
+  auto change3 =
+      MakeTargetChange(WatchTargetChangeState::Current, {1}, resume_token1_);
+
+  WatchChangeAggregator aggregator = CreateAggregator(
+      target_map, no_outstanding_responses_,
+      DocumentKeySet{doc1.key(), doc2.key()},
+      Changes(std::move(change1), std::move(change2), std::move(change3)));
+
+  RemoteEvent event = aggregator.CreateRemoteEvent(testutil::Version(3));
+
+  ASSERT_EQ(event.snapshot_version(), testutil::Version(3));
+  ASSERT_EQ(event.document_updates().size(), 2);
+  ASSERT_EQ(event.document_updates().at(doc1.key()), doc1);
+  ASSERT_EQ(event.document_updates().at(doc2.key()), doc2);
+
+  ASSERT_EQ(event.target_changes().size(), 2);
+
+  TargetChange target_change1{resume_token1_, true, DocumentKeySet{},
+                              DocumentKeySet{doc1.key(), doc2.key()},
+                              DocumentKeySet{}};
+  ASSERT_TRUE(event.target_changes().at(1) == target_change1);
+
+  TargetChange target_change2{resume_token1_, false, DocumentKeySet{},
+                              DocumentKeySet{}, DocumentKeySet{}};
+  ASSERT_TRUE(event.target_changes().at(2) == target_change2);
+
+  // The given BloomFilter will return false on MightContain(doc1) and true on
+  // MightContain(doc2).
+  ExistenceFilterWatchChange change4{
```

Comment:
this bloom filter proto is generated using the helper in web SDK, and copy paste the result here. Should i create a helper in iOS as well or this is okay?

Reply:
TODO: find a way to set databaseId here